### ISSUE TITLE
Accessibility: WCAG violation by different order of Blocks & Patterns in block inserter in Post- and Site-Editor

### ISSUE THREAD

**Issue by burnuser** [WordPress/gutenberg#47113]
### Description

As reported in core https://core.trac.wordpress.org/ticket/57431 and advised to report it here.
(And please don't play "not right located here" tennis!)

### Step-by-step reproduction instructions

In the block inserter of the post-editor are tabs ordered: Blocks - Patterns
In the block inserter of the site-editor are tabs ordered: Patterns - Blocks

It is confusing (and very bad UI design) for ALL users, when parts in the - expectable - same user interface change places.

And it is a direct violation of the "Web Content Accessibility Guidelines"
=> "Make Web pages appear and operate in predictable ways"
https://www.w3.org/TR/WCAG21/#consistent-navigation

### Environment info

WP 6.1.1
with and without Gutenberg

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by carolinan** [comment]
I don't think it is that simple. Patterns are highlighted in the site editor because there you are working mostly on design, as opposed to adding many single blocks. It might still very well be the case that consistency is more important.

**Comment by joedolson** [comment]
Regarding the WCAG violation question, I'm not sure this actually is a violation; it's debatable whether the site editor and the post editor should really be considered as the same interface. The two interfaces perform different functions, edit different fields, and have a *lot* of very different features, though the interfaces are similar.

However, I do think that consistency and muscle memory play a significant role in making the interface easier to use, and having these tools in a different order in the two editors does not seem beneficial to me. I'm not convinced that the argument "you are working mostly on design in the block editor" is sufficient to merit an otherwise arbitrary change in feature order.

**Comment by burnuser** [comment]
About same or not the same editor: at least the WordPress promotion advertises it as "same editor for posts and templates"

But of course more important is usability. And in my experience I work regularly with the post editor and have automated the tasks using block-inserter with Blocks at first and Patterns in the second tab.
And when I - sometimes - have something to do in the site-editor, the different order of blocks and patterns breaks every time my workflow!
My focus is on the content and should not be requested by changes in the common user interface!

And about priority of patterns or blocks:
1.) Maybe someone starts with a pattern (1 time), but all following steps of adaptation requires blocks (many times)!
2.) I think most people starts not with a fresh template, but want to change an existing one. And need blocks for this task!
3.) Nobody knows really, how the site-editor is used by other people. So, the need of consistency in the user interface for usability and accessibility is the only certainty.

**Comment by carolinan** [comment]
@WordPress/gutenberg-design I wanted to highlight this feedback for your consideration, not only for the positioning of the blocks and patterns tab panels, but also for [45722](https://github.com/WordPress/gutenberg/issues/45722).

**Comment by joedolson** [comment]
I think that changing the order because we're making assumptions about what users are doing or prioritizing is overstepping; we should offer an interface that is predictable, and that means keeping common items in the same order and location as much as we can. 

In my opinion, the consistent order should be Blocks > Patterns > Media in both environments.

**Comment by richtabor** [comment]
> In my opinion, the consistent order should be Blocks > Patterns > Media in both environments.

I agree. Consistency is key.

**Comment by carolinan** [comment]
I was going to start on this but found that it has already been solved. The order is the same in both editors. 
I tested on 6.4 RC 2.

**Comment by burnuser** [comment]
NO, it is NOT resolved!

When you start with website editor, you are in the regular page editing mode, where the order is Blocks > Patterns

But when you change to template editing mode, the order is CHANGED to Patterns > Blocks (even in 6.4 RC 2)

**Comment by burnuser** [comment]
And that the order changes in the same editor, dependent on the edited content, makes the confusion - obviously - even worse!

**Comment by joedolson** [comment]
Confirming. If you go to Editor > Pages and select a page, then open the canvas to edit, the order is Blocks / Patterns / Media.

If you go to Editor > Templates and select a template, then open the canvas to edit, the order is Patterns / Blocks / Media.

Reopening.

**Comment by joedolson** [comment]
@burnuser Please, do remember to try and be polite when communicating with other contributors here. We're all trying, and being shouted at is not constructive or helpful.

**Comment by alexstine** [comment]
Just going to add a quick 👍 here. UX should be simplified. Thanks.
